### PR TITLE
posix: fix ltp test errors

### PR DESCRIPTION
**Pull request by yangyalei**
### **Summary**
fix some posix ltp test errors

### **Impact**
unionfs

### **Testing**
sim

### CODE REVIEW

File: sched/pthread/pthread_create.c [hunk 1]
```diff
@@ -196,9 +196,13 @@ int nx_pthread_create(pthread_trampoline_t trampoline, FAR pthread_t *thread,
 
   if (!attr)
     {
-      /* Inherit parent priority by default */
+      /* Inherit parent priority by default. except idle */
+
+      if (parent->pid != IDLE_PROCESS_ID)
```
Comment (xiaoxiang781216) [inline sched/pthread/pthread_create.c:201]: call is_idle_task(parent) instead

File: mm/umm_heap/umm_malloc.c [hunk 2]
```diff
@@ -57,8 +57,16 @@ FAR void *malloc(size_t size)
...
 #else
+  FAR void *ret;
+
   /* Use mm_malloc() because it implements the clear */
 
-  return mm_malloc(USR_HEAP, size);
+  ret = mm_malloc(USR_HEAP, size);
+  if (!ret)
+    {
+      set_errno(ENOMEM);
```
Comment (xiaoxiang781216) [inline mm/umm_heap/umm_malloc.c:67]: other allocation function need set_errno too

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]
please squash the ltp: pathes

**Comment by yangyalei** [comment]
> please squash the ltp: patches

done @xiaoxiang781216

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**